main.tf

```diff
@@ -0,0 +1,84 @@
+locals {
+  id          = "${local.name}-${random_string.this.result}"
+  name        = var.name == null ? local.image_name : var.name
+  tags        = merge(var.tags, { Name = local.id })
+  image_repo  = replace(var.image, "/:.*/", "")
+  image_name  = replace(local.image_repo, "/.*//", "")
+  vpc_found   = length(data.aws_vpcs.this.ids) > 0
+  vpc_id      = local.vpc_found ? data.aws_vpcs.this.ids[0] : aws_vpc.this[0].id
```

Comment:
Have you seen this? I wonder if it could be helpful here:
https://developer.hashicorp.com/terraform/language/functions/try